### PR TITLE
MM-T248 Mark DM post as Unread

### PR DESCRIPTION
**Pull request by poflankov**
#### Summary
Add the tests for marking DM post as Unread in central panel and RHS.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/18166

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by hmhealey** [comment]
Thanks for the PR! This looks mostly good, but I'm not sure about the way we're chaining the Cypress calls since it's a different style than I've seen throughout the E2E tests I've seen. I'll defer to Saturn on that though since this is more his area.

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thank you @poflankov! Unfortunately, we're not going to use a different or mixed style in writing Cypress test at this stage, for some of the reasons:
- consistency in our codebase
- readability especially for non-technical reviewers (chaining is a bit overhead to them)

We use chaining for the built-in functions (`should`, `its`, `within`, etc) and it makes it easier to follow along but not when mixed up with custom commands.

Hope it's fine with you.

**Comment by poflankov** [comment]
@hmhealey,  @saturninoabril
I fixed the code style. Will it fit this way?

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @poflankov for updating! LGTM, except for few comments.

File: e2e/cypress/integration/mark_as_unread/helpers.js [hunk 1]
```diff
@@ -23,12 +23,12 @@ export function markAsUnreadShouldBeAbsent(postId, prefix = 'post', location = '
...
-    cy.get('.dropdown-menu').should('be.visible').within(() => {
-        cy.findByText('Mark as Unread').should('be.visible').click();
-    });
+export function markAsUnreadFromMenu(...args) {
+    cy.
+        clickPostDotMenu(...args).
+        get('.dropdown-menu').
+        findByText('Mark as Unread').
+        click();
 }
```
[the remark this answers is not in the document]
Comment (poflankov) [reply e2e/cypress/integration/mark_as_unread/helpers.js]: Fixed

File: e2e/cypress/integration/mark_as_unread/mark_dm_post_as_unread_spec.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            numberOfMessages: 3,
+            sender: this.mainUser,
+            channelId: this.dmChannel.id,
+        });
+
+        // # Visit the DM channel and open the thread in RHS
+        cy.apiLogin(this.mainUser).visit(this.link);
+
+        // # Мark the message from user A as unread
+        cy.then(() => markAsUnreadFromMenu(this.unreadFromHere.id));
```
[the remark this answers is not in the document]
Comment (poflankov) [reply e2e/cypress/integration/mark_as_unread/mark_dm_post_as_unread_spec.js]: Fixed

**Review by hmhealey — APPROVED** [review]
Yep, that looks good to me now. Thanks for changing all those!

**Review by saturninoabril — APPROVED** [review]
Thank you @poflankov! All looks good to me.

**Pull request merged**